staging/src/k8s.io/dynamic-resource-allocation/deviceclass/cache/cache.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"fmt"
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+	resourceapi "k8s.io/api/resource/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	resourceinformers "k8s.io/client-go/informers/resource/v1"
+	"k8s.io/client-go/tools/cache"
+	klog "k8s.io/klog/v2"
+)
+
+// ExtendedResourceCache maintains a global cache of extended resource to device class mappings
+// that can also be accessed by event handlers without requiring an extra scheduling cycle.
+type ExtendedResourceCache struct {
+	mutex sync.RWMutex
+	// mapping maps extended resource name to device class name
+	mapping map[v1.ResourceName]string
+	logger  klog.Logger
+}
+
+// NewExtendedResourceCache creates a new ExtendedResourceCache instance and populates it
+// with the current device classes from the informer cache.
+func NewExtendedResourceCache(deviceClassInformer resourceinformers.DeviceClassInformer, logger klog.Logger) *ExtendedResourceCache {
+	cache := &ExtendedResourceCache{
+		mapping: make(map[v1.ResourceName]string),
+		logger:  logger,
+	}
+
+	// Populate the cache with existing device classes from the informer
+	if deviceClassInformer != nil {
```

Comment:
it's easier to read by reduce the nesting

if deviceClassInformer == nil {
  return cache
}